core/src/test/kotlin/org/stellar/anchor/sep31/Sep31ServiceTest.kt

```diff
@@ -326,11 +351,84 @@ class Sep31ServiceTest {
     // TODO: Add fee validation.
     assertEquals("100", txn.amountIn)
     assertEquals(
-      "stellar:USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN",
-      txn.amountInAsset,
+        "stellar:USDC:GA5ZSEJYB37JRC5AVCIA5MOP4RHTM335X2KGX3IHOJAPP5RE34K4KZVN",
+        txn.amountInAsset,
     )
     assertEquals("12500", txn.amountOut)
     assertEquals(stellarJPYC, txn.amountOutAsset)
+
+    // Test null quote failure
+    Context.get().setQuote(null)
+    assertThrows<ServerErrorException> { sep31Service.updateTxAmountsBasedOnQuote() }
+  }
+  @Test
+  fun test_getTransaction() {
+    assertThrows<BadRequestException> { sep31Service.getTransaction(null) }
+    assertThrows<BadRequestException> { sep31Service.getTransaction("") }
+
+    every { txnStore.findByTransactionId("not_found") } returns null
+    assertThrows<NotFoundException> { sep31Service.getTransaction("not_found") }
```

Comment:
💡 Can you assert the exception class and message here too?

Reply:
The exception class is checked by `assertThrows<NotFoundException>`.

Since there is only one place where `NotFoundException` is thrown, I skipped checking the message. I had added the message check so that we won't miss it if we add more `NotFoundException` to the function. 🙂